gdb_backend/program.c

```diff
@@ -1,6 +1,270 @@
-int main() {
-    int a = 1;
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
 
+void func1(int *arr, int len, int modify);
+double slow_pow(double base, int exponent);
+double fast_pow(double base, int exponent);
 
-    int b = 2;
-}
\ No newline at end of file
+struct node {
+	int data;
+	struct node *next;
+};
+
+struct list_2 {
+    struct node *head;
+};
```

Comment:
name `list_2` a bit confusing, can we just leave it `list`?

Reply:
I think we are using list_2 to avoid a naming conflict.